### DISCUSSION TITLE
Remote debugging

### DISCUSSION THREAD

**Discussion by rrirower** [rokucommunity/vscode-brightscript-language#513] [Q&A]
I have a situation where I would like to debug some code running on a remote Roku (not in my local network).  Is that possible with the current Brighscript plugin for Visual Studio Code?

**Comment by TwitchBronBron** [comment]
Yes, it should technically be possible, assuming the remote Roku is in dev mode, is accessible from your current location.

For example, say you have a Roku at your house that you want to debug from a starbucks. On your home router, you would need to port-forward the following ports:
- port 80 (for roku app zip uploading) 
- port 8085 (for telnet console scraping)
- port 8060 (for remote presses)

If your ISP doesn't block any of these ports, then you can just forward the same port numbers. If not, then you might need to find unblocked ports. So then, for exapmle, you could do external: 8080 -> 192.168.1.30:80 (your roku's IP address). 

You'd also need to make sure you have the IP address of your house available. I use no-ip, but any dynamic DNS service would work. 

So then, putting it all together, your launch.json could look something like this:

```jsonc
{
	"type": "brightscript",
	"host": "your-custom-name.no-ip.biz",
	"password": "your-roku-dev-password",

         //change these ports if you had to use different external ports during port-forwarding
	"remotePort": 8060,
	"packagePort": 80,
        //this one is undocumented, but should work
	"brightScriptConsolePort": 8085,
}
```

I'm sure there are other ways, like perhaps a VPN or something like an [ngrok](https://ngrok.com/) or ssh tunnel.

**Reply by rrirower** [reply]
Thanks for this info. Very helpful.

**Reply by chrisdp** [reply]
@rrirower another option you can explore if you have computer in the network with your Roku is VS Code Server. 

https://code.visualstudio.com/docs/remote/vscode-server

This basically lets you connect your VS code on one computer and run a tunnel connection to another running the server. I have used this quite a bit and it's been really good.

**Reply by rrirower** [reply]
@chrisdp Thanks.  Definitely another option to consider.
